app/services/mediator.rb

```diff
@@ -1,40 +1,45 @@
-module Mediator
+class Mediator
   class IngestError < StandardError; end
   class DocumentHasNoFile < IngestError; end
-  class DocumentNotChunkable < IngestError; end
-  class ConversionUnimplemented < IngestError; end
+  class CannotIngestDocument < IngestError; end
+
+  attr_reader :document
 
   def self.ingest(document)
-    case next_step(document)
-    when :fetch
-      # Document has no file, so kick off fetching if it's a link
-      raise DocumentHasNoFile, "No file exists, file name is nil" unless document.web?
+    Mediator.new(document).execute
+  end
+
+  def initialize(document)
+    @document = document
+  end
 
+  def execute
+    case next_step
+    when :fetch
       FetchWebDocumentJob.perform_async(document.id)
     when :chunk
-      # Document has a file, so kick off chunking
       ChunkDocumentJob.perform_async(document.id)
     when :convert
-      # TODO: implement conversion to a chunkable format
-      raise ConversionUnimplemented, "Cannot convert: #{document.filename}, #{document.content_type}"
+      ConvertDocumentJob.perform_async(document.id)
     else
-      raise DocumentNotChunkable, "Cannot ingest document: #{document.filename}, #{document.content_type}"
+      raise CannotIngestDocument,
+        "Cannot ingest document: #{document.filename || 'nil'}, #{document.content_type || 'unknown'}"
     end
   end
 
-  def self.next_step(document)
+  private
+
+  def next_step
     if document.filename.nil?
-      :fetch
+      if document.web?
+        :fetch
+      end
     elsif document.file.attached?
-      if Parsers.textual?(document.content_type)
+      if Parsers.can_chunk?(document)
         :chunk
-      elsif need_conversion?(document)
+      elsif Converters.can_convert?(document)
         :convert
       end
     end
```

Comment:
Would this class be simpler if these conditionals were directly in `#next_step`?

Reply:
I don't understand ... do you mean run the jobs without returning the symbols for the case/when?

Reply:
Yes.

I may be missing something, but it looks like, of these two methods, one decides what to do and one does it. But, since "doing it" is just a one-liner, you could "do the thing" instead of telling the other method to do the thing.

Reply:
Because of the error handling, the single method versions that also raise an ingest error all result in complexity warnings by rubocop.